junifer/preprocess/confounds.py

```diff
@@ -163,63 +230,289 @@ def get_output_kind(self, input: List[str]) -> List[str]:
+
+        Returns
+        -------
+        to_select : list of str
+            List of confounds to select from the confounds file, based on the
+            strategy
+        squares_to_compute : dict
+            Dictionary containing the missing power2 confounds to compute
+            (keys) and the corresponding confounds to square (values)
+        derivatives_to_compute : dict
+            Dictionary containing the missing derivatives confounds to compute
+            (keys) and the corresponding confounds to compute the derivative
+            from (values)
+        spike_name : str
+            Name of the confound to use for spike detection
+
+        """
         confounds_df = input["data"]
-        confounds_spec = input["names"]["spec"]
-        # for every confound there is a derivative
-        # and for every confound + derivative there should be squares
-        derivatives_to_compute = input["names"].get("derivatives", {})
-        squares_to_compute = input["names"].get("squares", {})
-        spike_name = input["names"]["spike"]
+        available_vars = confounds_df.columns
+
+        # This function should build this 3 variables
+        to_select = []  # the list of confounds to select
+        squares_to_compute = {}  # the dictionary of missing squares
+        derivatives_to_compute = {}  # the dictionary of missing derivatives
+
+        for t_kind, t_strategy in self.strategy.items():
+            t_basics = FMRIPREP_BASICS[t_kind]
+
+            if any(x not in available_vars for x in t_basics):
+                missing = [x for x in t_basics if x not in available_vars]
+                raise ValueError(
+                    "Invalid confounds file. Missing basic confounds: "
+                    f"{missing}. "
+                    "Check if this file is really an fmriprep confounds file. "
+                    "You can also modify the confound removal strategy."
+                )
+
+            to_select.extend(t_basics)
+
+            if t_strategy in ["power2", "full"]:
+                for x in t_basics:
+                    x_2 = f"{x}_power2"
+                    to_select.append(x_2)
+                    if x_2 not in available_vars:
+                        squares_to_compute[x_2] = x
+
+            if t_strategy in ["derivatives", "full"]:
+                for x in t_basics:
+                    x_derivative = f"{x}_derivative1"
+                    to_select.append(x_derivative)
+                    if x_derivative not in available_vars:
+                        derivatives_to_compute[x_derivative] = x
+                    if t_strategy == "full":
+                        x_derivative_2 = f"{x_derivative}_power2"
+                        to_select.append(x_derivative_2)
+                        if x_derivative_2 not in available_vars:
+                            squares_to_compute[x_derivative_2] = x_derivative
+        spike_name = "framewise_displacement"
```

Comment:
above spike_name is a parameter in the function, but here it is defined/overwritten anyhow, or maybe am I missing something?

Reply:
it's not a parameter of the function. This function is to give the parameters to the confound prepare function. Basically, needs to find the variables to use, which squares/derivatives to compute and what is the name of the variable to use for searching "spikes".

Reply:
Ah sorry, I was looking at the docstring for `Returns`